tools/power_analysis/wpilog.py

```
@@ -0,0 +1,302 @@
+    "boolean": _decode_boolean,
+    "string": _decode_string,
+    "json": _decode_string,
+    "double[]": _decode_double_array,
+    "float[]": _decode_float_array,
+    "int64[]": _decode_int64_array,
+    "boolean[]": _decode_boolean_array,
+    "string[]": _decode_string_array,
+}
+
+
+def _decoder_for(type_name: str):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `_decoder_for`

(ANN202)